src/st/lib/e_library_bg.c

```diff
@@ -0,0 +1,277 @@
+static SVECTOR D_us_80180DD8 = {0, 0, 0};
+
+// parallax desk & chairs
+void EntityBackgroundFurniture(Entity* self) {
+    Primitive* prim;
+    s32 primIndex;
+    long p, flag;
+    s16 x;
+    long sxy;
+    SVECTOR svector;
+    VECTOR vector;
+    MATRIX matrix;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(g_EInitInteractable);
+        primIndex = g_api.AllocPrimitives(PRIM_GT4, 4);
+        if (primIndex != -1) {
+            self->flags |= FLAG_HAS_PRIMS;
+            self->primIndex = primIndex;
+            prim = &g_PrimBuf[primIndex];
+            self->ext.prim = prim;
+            while (prim != NULL) {
+                prim->tpage = 0xF;
+                prim->clut = 0x34;
+                prim->u0 = prim->u2 = 128;
+                prim->u1 = prim->u3 = 248;
+                prim->v0 = prim->v1 = 0;
+                prim->v2 = prim->v3 = 28;
+                prim->priority = 0x50;
+                prim->drawMode = DRAW_UNK02;
+                prim = prim->next;
+            }
+        } else {
+            DestroyEntity(self);
+            return;
+        }
+        // fallthrough
+    case 1:
+        SetGeomScreen(1024);
+        SetGeomOffset(128, self->posY.i.hi);
+        RotMatrix(&D_us_80180DD8, &matrix);
+        vector.vx = self->posX.i.hi - 128;
+        vector.vy = 0;
+        vector.vz = 1024;
+        TransMatrix(&matrix, &vector);
+        SetRotMatrix(&matrix);
+        SetTransMatrix(&matrix);
+        RotTransPers(&D_us_80180DD0, &sxy, &p, &flag);
+
+        x = sxy & 0xFFFF;
+        if (x > 0) {
+            while (x > -64) {
+                x -= 120;
+            }
+        } else {
+            while (x < -128) {
+                x += 120;
+            }
+        }
+
+        prim = self->ext.prim;
+        while (prim != NULL) {
+            prim->x0 = prim->x2 = x;
+            x += 120;
+            prim->x1 = prim->x3 = x;
+            prim->y0 = prim->y1 = self->posY.i.hi - 28;
+            prim->y2 = prim->y3 = self->posY.i.hi;
+            prim->drawMode = DRAW_UNK02;
+
+            prim = prim->next;
+            if (x > 256) {
+                break;
+            }
+        }
+
+        while (prim != NULL) {
+            prim->drawMode = DRAW_HIDE;
+            prim = prim->next;
+        }
+        break;
+    }
+}
+
+static SVECTOR D_us_80180DE0 = {0, 0, 448};
+static SVECTOR D_us_80180DE8 = {0, 0, 0};
+
+// parallax column shadow
+void EntityBackgroundShadow(Entity* self) {
+    Primitive *prim, *last;
+    s32 primIndex;
+    long p, flag;
+    s16 x;
+    long sxy;
+    SVECTOR svector;
+    VECTOR vector;
+    MATRIX matrix;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(g_EInitInteractable);
+        primIndex = g_api.func_800EDB58(PRIM_TILE_ALT, 10);
+        if (primIndex != -1) {
+            self->flags |= FLAG_HAS_PRIMS;
+            self->primIndex = primIndex;
+            prim = &g_PrimBuf[primIndex];
+            self->ext.prim = prim;
```

Comment:
Would this make sense as a `Primitive*` at 0x7C on `self->ext.libraryShadow`? Since that is stubbed anyway.
I'm not 100% clear on when we use `self->ext.prim` vs. not.

Reply:
I was using it out of convenience in the previous two functions to avoid creating a new extension struct, but should have used the LibraryShadow struct in that last function. Thanks for catching.